src/react/src/components/Contribute/SearchByOsIdTab.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+import InfoOutlinedIcon from '@material-ui/icons/InfoOutlined';
+
+import { makeSearchByOsIdTabStyles } from '../../util/styles';
+
+const SearchByOsIdTab = ({ classes }) => {
+    const [value, setValue] = useState('');
+    const history = useHistory();
+
+    const handleChange = event => {
+        const uppercaseValue = event.target.value.toUpperCase();
+        setValue(uppercaseValue);
+    };
+
+    const handleSearch = () => {
+        if (value.length === 15) {
```

Comment:
I suggest to move FE os id validation to `src/react/src/util/util.js`. Otherwise, `15` looks like a magic number for me.